components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/common/JwksHandler.java

```diff
@@ -0,0 +1,280 @@
+                ServiceReferenceHolder.getInstance().getOauthServerConfiguration().getIdTokenSignatureAlgorithm());
+        if (!accessTokenSignAlgorithm.equals(idTokenSignAlgorithm)) {
+            diffAlgorithms.add(idTokenSignAlgorithm);
+        }
+        JWSAlgorithm userInfoSignAlgorithm = mapSignatureAlgorithmForJWSAlgorithm(
+                ServiceReferenceHolder.getInstance().getOauthServerConfiguration().getUserInfoJWTSignatureAlgorithm());
+        if (!accessTokenSignAlgorithm.equals(userInfoSignAlgorithm)
+                && !idTokenSignAlgorithm.equals(userInfoSignAlgorithm)) {
+            diffAlgorithms.add(userInfoSignAlgorithm);
+        }
+        return diffAlgorithms;
+    }
+
+    /**
+     * This method generates the key store file name from the Domain Name.
+     *
+     * @return key store file name
+     */
+    private String generateKSNameFromDomainName(String tenantDomain) {
+        String ksName = tenantDomain.trim().replace(".", "-");
+        return (ksName + APIConstants.KeyStoreManagement.KEY_STORE_EXTENSION_JKS);
+    }
+
+    /**
+     * This method returns a set of certificates depending on the provided tenant domain.
+     *
+     * @param tenantDomain tenant domain which is used to get the relevant key store and extract certificates
+     * @return set of certificates
+     */
+    private Set<Certificate> getCertificates(String tenantDomain) {
+        Set<Certificate> certificates = new HashSet<>();
+        KeyStore keyStore;
+        try {
+            if (!APIConstants.SUPER_TENANT_DOMAIN.equals(tenantDomain)) {
+                // get tenant keyStore
+                int tenantId = APIUtil.getTenantIdFromTenantDomain(tenantDomain);
+                APIUtil.loadTenantRegistry(tenantId);
+                KeyStoreManager keyStoreManager = KeyStoreManager.getInstance(tenantId);
+                keyStore = keyStoreManager.getKeyStore(generateKSNameFromDomainName(tenantDomain));
+            } else {
+                // get super tenant keyStore
+                boolean tenantFlowStarted = false;
+                try {
+                    PrivilegedCarbonContext.startTenantFlow();
+                    PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                            .setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME, true);
+                    tenantFlowStarted = true;
+                    KeyStoreManager keyStoreManager = KeyStoreManager.getInstance(
+                            APIUtil.getTenantIdFromTenantDomain(APIConstants.SUPER_TENANT_DOMAIN));
+                    keyStore = keyStoreManager.getPrimaryKeyStore();
+                } finally {
+                    if (tenantFlowStarted) {
+                        PrivilegedCarbonContext.endTenantFlow();
+                    }
+                }
+            }
+            certificates.addAll(getCertificatesFromKeyStore(keyStore));
+        } catch (Exception e) {
```

Comment:
Bad practice to catch generic exception. see https://rules.sonarsource.com/java/RSPEC-2221/

Reply:
Seems like we can't remove the generic exception here as it is required by `getPrimaryKeyStore` method. Refer to the screenshot attached below:

<img width="1004" alt="Screenshot 2023-07-24 at 13 16 04" src="https://github.com/wso2-support/carbon-apimgt/assets/30475839/5aea895f-d8cf-4963-9313-86b27764d4c3">